packages/ckeditor5-ui/src/dialog/dialogview.ts

```diff
@@ -293,6 +298,15 @@ export default class DialogView extends DraggableViewMixin( View ) implements Dr
 		}
 	}
 
+	/**
+	 * TODO
+	 */
+	public override destroy(): void {
+		this.element?.remove();
```

Comment:
We've never done anything like this. It's the parent view or the controller that are responsible for removing views from DOM. I'd mirror `editor.ui.view.body.add( Dialog.view );` behavior in `Dialog` class and clean up things there.